### PR TITLE
CAMEL-19060: use LongAdder in DefaultInflightRepository to reduce contention

### PR DESCRIPTION
**Pull request by orpiske**
This seems to help (mostly?) consumers when there is heavy contention. For instance, with Disruptor on a 6 consumer / 3 producer thread configuration:

```
GEOMETRIC MEAN (RATE):
Test: 1041550.6594790532 | Baseline: 703258.587167714
Delta: 338292.0723113392
The geometric mean rate was greater for the test than for the baseline
```

Where "Test" is this patch and "Baseline" is 4.0.0-M3.

### CODE REVIEW

**Review by davsclaus — APPROVED** [review]

**Review by oscerd — APPROVED** [review]

**Comment by davsclaus** [comment]
I think this may break stuff, I can now see jbang have incorrect inflight exchanges

out in 5 seconds. Inflights per route: [route2 = 26]
2023-05-07 13:53:14.719  INFO 39678 --- [ - ShutdownTask] impl.engine.DefaultShutdownStrategy : Waiting as there are still 26 inflight and pending exchanges to complete, timeout in 4 seconds. Inflights per route: [route2 = 26]
2023-05-07 13:53:15.721  INFO 39678 --- [ - ShutdownTask] impl.engine.DefaultShutdownStrategy : Waiting as there are still 26 inflight and pending exchanges to complete, timeout in 3 seconds. Inflights per route: [route2 = 26]
2023-05-07 13:53:16.727  INFO 39678 --- [ - ShutdownTask] impl.engine.DefaultShutdownStrategy : Waiting as there are still 26 inflight and pending exchanges to complete, timeout in 2 seconds. Inflights per route: [route2 = 26]
2023-05-07 13:53:17.731  INFO 39678 --- [ - ShutdownTask] impl.engine.DefaultShutdownStrategy : Waiting as there are still 26 inflight and pending exchanges to complete, timeout in 1 seconds. Inflights per route: [route2 = 26]

**Comment by davsclaus** [comment]
Yes this works fine on 3.20.x branch

**Comment by orpiske** [comment]
> Yes this works fine on 3.20.x branch

How about with M3?

File: core/camel-base-engine/src/main/java/org/apache/camel/impl/engine/DefaultInflightRepository.java [hunk 1]
```diff
@@ -67,28 +67,28 @@ public void remove(Exchange exchange) {
...
         }
     }
 
     @Override
     public void remove(Exchange exchange, String routeId) {
-        AtomicInteger existing = routeCount.get(routeId);
+        LongAdder existing = routeCount.get(routeId);
         if (existing != null) {
-            existing.decrementAndGet();
+            existing.increment();
```
Comment (orpiske) [inline core/camel-base-engine/src/main/java/org/apache/camel/impl/engine/DefaultInflightRepository.java:80]: @davsclaus i think the bug could be here. It should decrement, but it is incrementing.

**Pull request merged**